### DISCUSSION TITLE
extractVersion doesn't seem to be working for docker tags with a release prefix

### DISCUSSION THREAD

**Discussion by rcdailey** [renovatebot/renovate#30072] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I have the following package rule, which was taken from the `extractVersion` docs (my full `renovate.json5` file can be found [here](https://github.com/rcdailey/nezuko/blob/master/renovate.json5)):

```json
    {
      "matchPackagePrefixes": ["ghcr.io/hotio/"],
      "extractVersion": "^release-(?<version>.*)$"
    },
```

I use this to support version tags in the format of `release-1.2.3`. All docker images starting with `ghcr.io/hotio/` will use this format consistently.

Here are the job logs I see:

```txt
DEBUG: Dependency ghcr.io/hotio/plex has unsupported/unversioned value release-1.40.2.8395-c67dce28e (versioning=docker)
DEBUG: Skipping ghcr.io/hotio/plex because no currentDigest or pinDigests
DEBUG: Dependency ghcr.io/hotio/sabnzbd has unsupported/unversioned value release-4.3.1 (versioning=docker)
DEBUG: Skipping ghcr.io/hotio/sabnzbd because no currentDigest or pinDigests
DEBUG: Dependency ghcr.io/hotio/qbittorrent has unsupported/unversioned value release-4.6.4 (versioning=docker)
DEBUG: Skipping ghcr.io/hotio/qbittorrent because no currentDigest or pinDigests
DEBUG: Dependency ghcr.io/hotio/prowlarr has unsupported/unversioned value release-1.16.2.4435 (versioning=docker)
DEBUG: Skipping ghcr.io/hotio/prowlarr because no currentDigest or pinDigests
DEBUG: Dependency ghcr.io/hotio/sonarr has unsupported/unversioned value release-4.0.4.1491 (versioning=docker)
DEBUG: Skipping ghcr.io/hotio/sonarr because no currentDigest or pinDigests
DEBUG: Dependency ghcr.io/hotio/sonarr has unsupported/unversioned value release-4.0.4.1491 (versioning=docker)
DEBUG: Skipping ghcr.io/hotio/sonarr because no currentDigest or pinDigests
DEBUG: Dependency ghcr.io/hotio/radarr has unsupported/unversioned value release-5.4.6.8723 (versioning=docker)
DEBUG: Skipping ghcr.io/hotio/radarr because no currentDigest or pinDigests
DEBUG: Dependency ghcr.io/hotio/radarr has unsupported/unversioned value release-5.4.6.8723 (versioning=docker)
DEBUG: Skipping ghcr.io/hotio/radarr because no currentDigest or pinDigests
DEBUG: Dependency ghcr.io/hotio/radarr has unsupported/unversioned value release-5.4.6.8723 (versioning=docker)
DEBUG: Skipping ghcr.io/hotio/radarr because no currentDigest or pinDigests
DEBUG: Dependency ghcr.io/hotio/tautulli has unsupported/unversioned value release-2.13.4 (versioning=docker)
DEBUG: Skipping ghcr.io/hotio/tautulli because no currentDigest or pinDigests
DEBUG: Dependency ghcr.io/hotio/overseerr has unsupported/unversioned value release-1.33.2 (versioning=docker)
DEBUG: Skipping ghcr.io/hotio/overseerr because no currentDigest or pinDigests
DEBUG: Dependency ghcr.io/hotio/unpackerr has unsupported/unversioned value release-0.13.1 (versioning=docker)
DEBUG: Skipping ghcr.io/hotio/unpackerr because no currentDigest or pinDigests
```

For some reason, the `extractVersion` setting isn't kicking in and I don't know why. What changes do I need to make?

### Logs (if relevant)

https://developer.mend.io/github/rcdailey/nezuko/-/job/01908fa3-fc1e-74cb-a330-f5b7bb5132fa

**Comment by RahulGautamSingh** [comment]
You misunderstoood `extractVersion`. It extracts version from the releases that are fetched and not the version from the package file.

What you need is a [custom regex manager](https://docs.renovatebot.com/modules/manager/regex/).

For example:
```js
{
  "customManagers": [{
    "customType": "regex",
    "fileMatch": ["docker-compose.yml"],
    "matchStrings": ["image: (?<depName>ghcr.io/hotio/.*):release-(?<currentValue>.*)"],
    "extractVersionTemplate": "^release-(?<version>.*)$",
    "datasourceTemplate": "docker"
  }]
}
```

**Reply by rcdailey** [reply]
What do you mean by releases that are fetched? Are you referring to GitHub releases?

**Reply by RahulGautamSingh** [reply]
Yes.

**Reply by rcdailey** [reply]
Looking back at the docs, I do see now that there's mention of "github release" before the examples &mdash; but clearly I missed that. The initial one-liner doesn't make that clear, in my opinion:

> A regex (re2) to extract a version from a datasource's raw version string.

I guess the question is, what does "datasource" mean? That is unclear to me. But when I see "raw version string," it makes me think of package versions (docker tags in this case).

Maybe there's some worthwhile clarification possible in the docs. I appreciate your help!

**Comment by rarkins** [comment, marked as answer]
Alternatively try using versionCompatibility

**Reply by rcdailey** [reply]
Thank you. In my case, what rule of thumb should I follow to decide between `versionCompatibility` and custom regex manager?

I will say just looking at the regex solution, it's very verbose and requires some confusing, extra configuration. Even though there's no "compatibility" data to pull from the tag in my case, it's the simpler solution?

**Reply by RahulGautamSingh** [reply]
If you are only concerned with extracting version from the raw version string -> `versionCompatiblity`
If you need more features that that -> `customManager`

For your case, `versionComptatibility` is the perfect solution.
